### ISSUE TITLE
Connect to IB TWS from multiple instance of matlab

### ISSUE THREAD

**Issue by manishranjan99** [softwarespartan/IB4m#3]
Hi Abel,

Thank you again for your great work!  I have run into issue where I can not connect the IB4m through multiple instances of Matlab.  For example, if I have a function in batch mode that is trying to connect while it is already connected to another function.  You can test it by opening two matlabs and try to connect through each.  Once it is connected to one, if throws java.eof.exception error into other.  

Would you recommend a workaround for this?  Always appreciate any help I can get on this.

Kind Regards,
Manish

**Comment by softwarespartan** [comment]
Hi Manish,

Great to hear from you.

Off the top of my head, you'll want to make sure that the CLIENTID in the session.eConnect call are unique.  Typically, in matlab just use client id of "0", so you want to increment that Id for each matlab instance.  

See link: https://www.interactivebrokers.com/en/software/api/apiguide/java/econnect.htm

I'll look into this more tonight or tomorrow morning when I'm back at my computer.

Cheers
-abel

**Comment by softwarespartan** [comment]
Hi Manish

I confirmed with multiple instances of matlab that as long as the CLIENT_ID is different for each matlab instance there does not appear to be any issues with multiple TWS connections.

Don’t hesitate to follow up if for some reason it’s still not working on your end with unique client id’s.  I’m happy to look into it some more. 

Cheers
-abel

**Comment by manishranjan99** [comment]
Hi Abel,

It does work with different client ID.  Thank you so much you for your help again.  

Kind Regards,
Manish

**Comment by manishranjan99** [comment]
Hi Abel,

I appreciate your help.  If you don't mind me asking, are you a matlab IB
algo trader?  I am still a newbie, working on a small gui based trading
routine in matlab and wondering if you have any experience in this.   .

Manish

**Comment by softwarespartan** [comment]
Hi Manish

I have a lot of matlab experience both as a trader and scientist.

What can I help you with?

Cheers
-abel

**Comment by manishranjan99** [comment]
Hi Abel,

How is your overall experience trading with matlab-IB?  Are there lot of
retail traders using it?  What were some of the challenges you faced during
trading with matlab-IB?   Is there any disadvantage of using matlab-IB?  Do
you use a fully automated trading with matlab-IB interface or
semiautomated?  I am working on a matlab ui based system which I hope can
make it fully automated one day.  It uses historic data from IB,
reqrealtime 5s bars from IB for trading.  At this point I haven't fully
implemented the who thing but I have some crude setup for paper trading.  I
am still implementing checks and double checks to foolproof my system but
it is still very crude at this stage.  I have a few technical indicator
based trading strategy in mind and few options combo based but I haven't
done much of back testing as yet.  What kind of strategy would you
recommend?  Do you have any blogs?
Sorry for asking too many questions.  Please take your time to answer it.

Regards,
Manish

**Comment by softwarespartan** [comment]
Hi Manish

Sounds like many of the typical growing pains of getting going in algo trading. 

I went through similar experiences in the beginning.  

There are a lot of great books out there.  I got started with books by Ernie Chan and Jack Schwager (Market Wizards!!).  

Most important is to find your own groove. Develop strategy that resonates with your personality and tolerance levels for risk/reward.

Backtesting is really important.  Don’t skimp on backtesting!  I’v coded up some strategies that look great on paper but fall flat as a real algorithm. 

Go slow and focus on building good “sea legs” and your intuition for the market.

I don’t have a blog … but maybe I’ll start one :)

Cheers
-abel
